### PR TITLE
[SYCL][E2E] Removing deprecated warnings in basic accessor, buffer, and queue e2e tests

### PR DESCRIPTION
**Pull request by ayylol**

### CODE REVIEW

File: sycl/test-e2e/Basic/buffer/buffer.cpp [hunk 1]
```diff
@@ -656,17 +656,15 @@ int main() {
     sycl::buffer<char, 1> Buf_2(Size / 2);
 
     {
-      auto AccA = Buf_1.get_access<sycl::access::mode::read_write>(Size / 2);
-      auto AccB = Buf_2.get_access<sycl::access::mode::read_write>(Size / 2);
+      auto AccA = Buf_1.get_host_access(Size / 2, sycl::read_write);
+      auto AccB = Buf_2.get_host_access(Size / 2, sycl::read_write);
```
Comment (steffenlarsen) [inline sycl/test-e2e/Basic/buffer/buffer.cpp:660]: Nit; AFAIK, though this is completely valid, the preferred way of constructing an accessor is using the deduction guides, i.e.

```suggestion
      sycl::host_accessor AccA{Buf_1, Size / 2, sycl::read_write};
      sycl::host_accessor AccB{Buf_2, Size / 2, sycl::read_write};
```

File: sycl/test-e2e/Basic/buffer/buffer_full_copy.cpp [hunk 2]
```diff
@@ -29,8 +29,8 @@ void check_copy_device_to_host(sycl::queue &Queue) {
   {
     sycl::range<2> acc_range(2, 2);
     sycl::id<2> acc_offset(offset, offset);
-    auto acc = simple_buffer.get_access<sycl::access::mode::read_write>(
-        acc_range, acc_offset);
+    auto acc =
+        simple_buffer.get_host_access(acc_range, acc_offset, sycl::read_write);
```
Comment (steffenlarsen) [inline sycl/test-e2e/Basic/buffer/buffer_full_copy.cpp:33]: Same as previous comment.

File: sycl/test-e2e/Basic/buffer/reinterpret.cpp [hunk 3]
```diff
@@ -101,12 +101,19 @@ int main() {
...
+      sycl::buffer<float, 1> buf_fl(r1d);
+      auto buf_d = buf_1d.reinterpret<double>(r2d);
+    } catch (sycl::exception e) {
+      if (e.code() == sycl::errc::invalid) {
+        std::cout << "Expected exception has been caught: " << e.what()
+                  << std::endl;
+        gotException = true;
+      }
+    }
+    assert(gotException);
```
Comment (steffenlarsen) [inline sycl/test-e2e/Basic/buffer/reinterpret.cpp:116]: Nit; Currently, we do not know the difference between an unexpected exception being thrown and no exception being thrown. Following would address this.
```suggestion
      assert(e.code() == sycl::errc::invalid);
      std::cout << "Expected exception has been caught: " << e.what()
                << std::endl;
      gotException = true;
    }
    assert(gotException);
```

File: sycl/test-e2e/Basic/buffer/subbuffer.cpp [hunk 4]
```diff
@@ -153,79 +153,111 @@ void checkExceptions() {
...
 
   // non-contiguous region
   try {
     sycl::buffer<int, 2> sub_buf{buf2d, /*offset*/ sycl::range<2>{2, 0},
                                  /*size*/ sycl::range<2>{2, 2}};
-    assert(!"non contiguous region exception wasn't caught");
-  } catch (const sycl::invalid_object_error &e) {
-    std::cerr << e.what() << std::endl;
+  } catch (const sycl::exception &e) {
+    if (e.code() == sycl::errc::invalid) {
```
Comment (steffenlarsen) [inline sycl/test-e2e/Basic/buffer/subbuffer.cpp:163]: Same as previous comment.

**Review by steffenlarsen — APPROVED** [review]
LGTM overall! Small nits.

**Comment by aelovikov-intel** [comment]
Looks like `SYCL :: Basic/buffer/buffer_create.cpp` is failing on Arc runner in post-commit after this. @ayylol , @againull .

**Comment by steffenlarsen** [comment]
Fix for buffer_create in https://github.com/intel/llvm/pull/13982.

**Pull request merged**